### ISSUE TITLE
cc_courses.schema.json

### ISSUE THREAD

**Issue by jheim2** [TIK-NFL/campusconnect#1]
`bwilias/campusconnect_schemas/cc_courses.schema.json`

* wäre es sinnvoll term und vielleicht year oder sowas zusätzlich als integer mitzuschleifen? Wir hatten am Anfang der Kopplung das Problem, dass die LectureID, die wir aus HIS bekommen hatten nicht eindeutig war. Die ändert sich über die Semester nicht. Wir haben es jetzt so gelöst, indem wir der LectureID 202201 für 2022 Sommersemester voranstellen. Aber potentiell könnten wir das vielleicht auch über ein zusätzliches Feld tun?
* `groupScenario` kann man bzw. soll man definieren, dass sich das eigentlich nicht ändern darf bzw soll? Also was passiert eigentlich wenn bei einer zweiten Syncronisation sich das groupScenario ändert? Kann oder sollte man da in den Schemas festschreiben darf sich nicht ändern?
* `linkToCurriculum` ist hier der Link ins Campusmanagement gemeint? Ist ein Link zurück immer ausreichend oder kann es auch den Fall geben, dass man mehre Links braucht? Je nach Campusmanagement-System vielleicht bei Szenario "Alle-Gruppen-in-einen-Kurs"? In HISinOne ist das glaube ich egal weil der Link dann auf die Veranstaltung mit den Gruppen gehen könnte.
* bei `linkToLecture` kann man mehrere Links setzen? Oder müsste das dann type "array" sein?
* hier haben die Gruppen id aber im course_member nicht und deswegen wird durchgezählt?

**Comment by bogenschlag** [comment]
> `groupScenario` kann man bzw. soll man definieren, dass sich das eigentlich nicht ändern darf bzw soll? Also was passiert eigentlich wenn bei einer zweiten Syncronisation sich das groupScenario ändert? Kann oder sollte man da in den Schemas festschreiben darf sich nicht ändern?

Hmm. Ich kann mir zumindest nicht vorstellen, wie das über das JSON gehen soll. Eigentlich ist das ein CMS-Problem. Da muss halt sichergestellt werden, dass man das nicht ändern kann.

**Comment by bogenschlag** [comment]
> `linkToCurriculum` ist hier der Link ins Campusmanagement gemeint? Ist ein Link zurück immer ausreichend oder kann es auch den Fall geben, dass man mehre Links braucht? Je nach Campusmanagement-System vielleicht bei Szenario "Alle-Gruppen-in-einen-Kurs"? In HISinOne ist das glaube ich egal weil der Link dann auf die Veranstaltung mit den Gruppen gehen könnte.

Ja, damit ist ein Link ins CMS gemeint. Curriculum ist ja tendenziell etwas oberhalb der Lehrveranstaltungsebene. Ob einer ausreicht oder ob wir das eher ganz rauskicken, weil das gar nicht nicht sinnvoll ist, weiß ich nicht. Muss man vielleicht mal mit CMS-Leuten drüber reden. (Unsere durften damals™ teilweise nicht zum Spielen rauskommen, als wir das gebaut haben …)

**Comment by bogenschlag** [comment]
> bei `linkToLecture` kann man mehrere Links setzen? Oder müsste das dann type "array" sein?

Nö, das ist ja der Link zurück zur Lehrveranstaltung im CMS. Da ein Kurs immer nur aus einer LV hervorgehen kann, brauchen wir hier auch nur einen Link IMHO.

**Comment by bogenschlag** [comment]
> hier haben die Gruppen id aber im course_member nicht und deswegen wird durchgezählt?

Du hängst dich zu sehr an Begrifflichkeiten auf. 😉 Wir waren damals einfach nicht 100%ig konsequent, sprich, hier es ID, drüber NUM, gemeint war/ist aber immer das selbe, die Nummer aus LSF. 😇
